programs/github

```diff
@@ -21,14 +21,62 @@ function progressbar(f)
 	return left .. right
 end
 
+-- get flag arguments
+local function findFlag(arg)
+  local accepted = {
+    { name = 'branch', flags = { '-b', '--branch' } },
+    { name = 'tag', flags = { '-t', '--tag' } },
+    { name = 'auth', flags = { '-a', '--auth' } }
+  }
+  local isFlag, flagName = false, nil
+  if arg:sub(1,1) == '-' or arg:sub(1,1) == '--' then
+    for _, flagType in pairs(accepted) do
+      for _, flag in pairs(flagType.flags) do
+        if arg == flag then
+          isFlag = true
+          flagName = flagType.name
+          return isFlag, flagName
+        end
+      end
+    end
+    if not isFlag then
+      return printError('Unknown flag argument '..arg)
+    end
+  end
+  return isFlag, flagName
+end
+
+local passed = {...}
+local args = {}
+local flags = {}
+local flagValue = ''
+for k, arg in pairs(passed) do
+  local isFlag, flagName = findFlag(arg)
+  if isFlag then
+    flags[flagName] = passed[k + 1]
```

Comment:
This gets unhappy on `github clone user/repo -b -b dest`, which should check out the branch `-b`